packages/terra-clinical-embedded-content-consumer/src/EmbeddedContentConsumer.jsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Consumer } from 'xfc';
+import styles from './EmbeddedContentConsumer.scss';
+
+const propTypes = {
+  /**
+   * The source URL of the content to load.
+   */
+  src: PropTypes.string.isRequired,
+  /**
+  * A set of event handlers keyed by the event name.
+  */
+  eventHandlers: PropTypes.arrayOf(PropTypes.shape({
+    key: PropTypes.string,
+    handler: PropTypes.func,
+  })),
+};
+
+class EmbeddedContentConsumer extends React.Component {
+
+  componentDidMount() {
+    Consumer.init();
+    this.xfcFrame = Consumer.mount(this.embeddedContentWrapper, this.props.src);
+
+    if (this.props.eventHandlers) {
+      this.props.eventHandlers.forEach((event) => {
```

Comment:
So something like this?
```jsx
<EmbeddedContentConsumer
  src="www.example.com"
  onMount={() => { console.log('mounted'); }}
  onLaunch={() => { console.log('launched'); }}
  onAuthorize={() => { console.log('authorized'); }}
  onFullscreen={() => { console.log('provider wants to go fullscreen'); }}
  customEvents={{
    customEvent1: () => { console.log('customEven1 was triggered'); }
    customEvent2: () => { console.log('customEven2 was triggered'); }
  }}
```
I'm not sure how we can get away with having a single customEvent handler (unless we could do a wildcard-like bind on the frame to catch everything that's not an xfc-specific event).